### PR TITLE
Added entity and repository for Announcement

### LINKED ISSUE #25: #15 Add announcements to commons play page.  Add Backend entity/repository for announcements. Issue B1
### Have you updated the title?

- [ ] I have updated the issue's title.

### As a...

admin

### I can...

add an announcement at the top of the commons

### So that...

I can communicate to the players

### Discussion

The announcements box at the top of the commons play page is intended for the administrator of the game to post announcements for players.   However, that feature hasn't been implemented yet.

This Epic (collection of stories) introduces that feature to the game.  You should break this epic into multiple issues before working on it.

Copy the user stories/acceptance criteria into a separate  issue before starting to work on it, and make PRs for each part separately.  Benefits to this staged approach:
* You'll earn more points that way
* You'll earn them more quickly
* PRs are easier to code review and merge
* If you don't finish, you get partial credit for what you did complete, and students in a future class can finish off your work.

The issues below are broken up into:
* two backend issues that must be completed in sequence: first B1 then B2.
* two frontend issues that must be worked in in sequence, first F1 then F2.  F1 can proceed in parallel with B1, B2 as long as the folks working on B2 and F1 coordinate the format of the fixtures.
* issue F2 depends on B2 and F1 being completed first.

# Issue B1:  Backend, Add Entity / Repository for announcements

Developer Story:

* As a developer
* I have access to a database table for announcements
* So that I can build future features using this table

Acceptance Criteria:

- [ ] There is an entity file for an Announcements table with field described below
- [ ] There is an repository file for a Announcements table 

Fields:
* id, long autogenerated
* commonsId, required, id of the associated commons
* start (required, datetime to start showing the announcement, defaults to current datetime)
* end (optional, datetime to stop showing the announcement)
* announcement (String)

### PR DESCRIPTION
**Pull request by wyatthspivak**
## Overview

In this PR, I added an Announcement entity class and a AnnouncementRepository for the first part of creating backend support for a Announcements in the commons play page

## Feedback Request (Optional)

Please give suggestions on the following:
- Do the entity fields I added line up with the ones in the EPIC issue #15 ?

## Future Possibilities (Optional)

Some possible points could be:
- This is the beginning/part of having announcements in the commons play page

## Validation (Optional)

1. Download this branch.
2. Run the project.
3. Compare in code the entity fields specified in the issues 

## Tests

- [ ] Backend Unit tests (`mvn test`) pass
- [ ] Backend Test coverage (`mvn test jacoco:report`) 100%
- [ ] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [ ] Frontend Unit tests (`npm test`) pass
- [ ] Frontend Test coverage (`npm run coverage`) 100%
- [ ] Frontend Mutation tests (`npx stryker run`) 100% 
- [ ] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #25

### CODE REVIEW

**Review by KarimBenyassine1 — DISMISSED** [review]
LGTM!

**Review by kaitntom — DISMISSED** [review]
LGTM!

File: src/main/java/edu/ucsb/cs156/happiercows/repositories/AnnouncementRepository.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package edu.ucsb.cs156.happiercows.repositories;
+
+import org.springframework.data.repository.CrudRepository;
+
+import edu.ucsb.cs156.happiercows.entities.Announcement;
+
+public interface AnnouncementRepository extends CrudRepository<Announcement, Long> {
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/repositories/AnnouncementRepository.java:8]: This needs to be annotated with the `@Repository` annotation (see the other repository classes in the same directory.

File: src/main/java/edu/ucsb/cs156/happiercows/repositories/AnnouncementRepository.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package edu.ucsb.cs156.happiercows.repositories;
+
+import org.springframework.data.repository.CrudRepository;
+
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/repositories/AnnouncementRepository.java]:
```suggestion
import org.springframework.stereotype.Repository;
```

File: src/main/java/edu/ucsb/cs156/happiercows/repositories/AnnouncementRepository.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+import edu.ucsb.cs156.happiercows.entities.Announcement;
+
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/repositories/AnnouncementRepository.java:6]:
```suggestion

@Repository
```

**Review by pconrad — CHANGES_REQUESTED** [review]
One small change needed.

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**